### ISSUE TITLE
Haskell Starter Pack

### ISSUE THREAD

**Issue by amstan** [aichallenge/aichallenge#66]
@jmcarthur has been working on a starter pack at http://darcsden.com/jmcarthur/ants-bot, but unfortunatelly he cannot continue working on it anymore.

Anyone wanting a haskell pack could continue working on that code, or use it for inspiration for a rewrite(according to @jmcarthur it's over-engineered).

**Comment by amstan** [comment]
@shangaslammi I noticed you have a haskell starter pack in progress, would you be interested in submitting it?

**Comment by shangaslammi** [comment]
@amstan: I'll gladly contribute my Haskell starter pack, but it should probably be reviewed by someone with more Haskell experience. I'm pretty new to Haskell myself, so I'm not sure about the code quality.

**Comment by jwkvam** [comment]
Hello, I wrote up a Haskell starter bot, https://github.com/qacek/ants-haskell. It mostly follows the current starter bot guide in the wiki plus some input I received from McLeopold on upcoming changes in the engine like dead ants' vision for 1 turn. The starter bot implements the basic N,E,S,W path finding algorithm. For better or worse I wrote it from scratch since I didn't see this issue until recently. I'm also new to Haskell so it would be nice to get feedback. It's still a work in progress but it should be usable enough in its current state.

By the way I'm testing with ghc 7.0.3, I don't know if it will be a problem if the contest machines are using a different version.
